provisioning/utils.go

```diff
@@ -48,21 +49,24 @@ func getPodHostIP(podClient coreclientv1.PodsGetter, targetNamespace string) (st
 	return hostIP, err
 }
 
-func GetServerInternalIP(osclient osclientset.Interface) (string, error) {
+func getServerInternalIP(client kubernetes.Interface, targetNamespace string, config *metal3iov1alpha1.ProvisioningSpec, osclient osclientset.Interface) (string, error) {
 	infra, err := osclient.ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
 	if err != nil {
 		err = fmt.Errorf("Cannot get the 'cluster' object from infrastructure API: %w", err)
 		return "", err
 	}
-
-	// NOTE(dtantsur): do we need to handle non-BareMetal platforms here?
-	if infra.Status.PlatformStatus.BareMetal == nil {
-		err = fmt.Errorf("Cannot detect server API VIP: not a baremetal platform")
+	// FIXME(dtantsur): handle the new APIServerInternalIPs field and the dualstack case.
+	switch infra.Status.PlatformStatus.Type {
+	case osconfigv1.BareMetalPlatformType:
+		return infra.Status.PlatformStatus.BareMetal.APIServerInternalIP, nil
+	case osconfigv1.OpenStackPlatformType:
+		return infra.Status.PlatformStatus.OpenStack.APIServerInternalIP, nil
+	case osconfigv1.VSpherePlatformType:
+		return infra.Status.PlatformStatus.VSphere.APIServerInternalIP, nil
+	default:
```

Comment:
Let's return "" in case of platform None and fall back to the pod IP.

Reply:
Thank you Dmitry, fixed now.